### PR TITLE
Add ccache to all jenkins images

### PR DESCRIPTION
**Pull request by ramereth**
This also pulls in some standardization on the Dockerfile and how we build these images. Please see individual commits for more details.

### CODE REVIEW

**Comment by ramereth** [comment]
[#29980: pytorch community request : cache folder for docker container](https://trello.com/c/OxnhKak7/699-29980-pytorch-community-request-cache-folder-for-docker-container)

File: centos-ppc64le-cuda/Dockerfile [hunk 1]
```diff
@@ -13,32 +21,82 @@ ENV JENKINS_AGENT_HOME ${JENKINS_AGENT_HOME}
...
-# install git
-RUN yum install -y git && yum clean all
+RUN groupadd -g ${gid} ${group} && \
+    useradd -d "${JENKINS_AGENT_HOME}" -u "${uid}" -g "${gid}" -m -s /bin/bash "${user}"
 
 # setup sudo
-RUN yum install -y sudo && yum clean all
-RUN echo "jenkins ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+RUN yum install -y sudo && yum clean all && \
+    yum clean all && \
```
Comment (Codysseus) [inline centos-ppc64le-cuda/Dockerfile]: `yum clean all` twice?

File: centos-ppc64le-cuda/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+ARG VERSION="9.0-devel-centos7"
+FROM nvidia/cuda-ppc64le:${VERSION}
 LABEL MAINTAINER="OSU Open Source Lab <dockerhub@osuosl.org>"
 
+# Point to our mirrors and update to latest
+COPY CentOS-Base.repo /etc/yum.repos.d/CentOS-Base.repo
+COPY epel.repo /etc/yum.repos.d/epel.repo
+RUN yum upgrade -y && \
+    yum clean all && \
+    rm -rf /var/cache/yum
```
Comment (Codysseus) [inline centos-ppc64le-cuda/Dockerfile:10]: This is what I get when I try to build it:
```
Step 6/30 : RUN yum upgrade -y &&     yum clean all &&     rm -rf /var/cache/yum
 ---> Running in 194d9572b956
standard_init_linux.go:185: exec user process caused "no such file or directory"
```

**Review by Codysseus — CHANGES_REQUESTED** [review]

File: centos-ppc64le-cuda/Dockerfile (hunk 2 above)
Comment (Codysseus) [reply centos-ppc64le-cuda/Dockerfile:10]: I ran this on openpower8 and it built, but now I can't find the container.

**Comment by ramereth** [comment]
@Codysseus can you please take another look at this?

**Review by Codysseus — CHANGES_REQUESTED** [review]
It seems the s390x builds have problems :/
```
Step 11/28 : RUN groupadd -g ${gid} ${group}     && useradd -d "${JENKINS_AGENT_HOME}" -u "${uid}" -g "${gid}" -m -s /bin/bash "${user}"
 ---> Running in c2f68355e6fe
standard_init_linux.go:190: exec user process caused "exec format error"
The command '/bin/sh -c groupadd -g ${gid} ${group}     && useradd -d "${JENKINS_AGENT_HOME}" -u "${uid}" -g "${gid}" -m -s /bin/bash "${user}"' returned a non-zero code: 1
[root@openpower8 debian-s390x]# 

Step 11/29 : RUN groupadd -g ${gid} ${group}     && useradd -d "${JENKINS_AGENT_HOME}" -u "${uid}" -g "${gid}" -m -s /bin/bash "${user}"
 ---> Running in 83906727b5ae
standard_init_linux.go:190: exec user process caused "exec format error"
The command '/bin/sh -c groupadd -g ${gid} ${group}     && useradd -d "${JENKINS_AGENT_HOME}" -u "${uid}" -g "${gid}" -m -s /bin/bash "${user}"' returned a non-zero code: 1
[root@openpower8 fedora-s390x]# 

Step 11/28 : RUN groupadd -g ${gid} ${group}     && useradd -d "${JENKINS_AGENT_HOME}" -u "${uid}" -g "${gid}" -m -s /bin/bash "${user}"
 ---> Running in 23d0742e662a
standard_init_linux.go:190: exec user process caused "exec format error"
The command '/bin/sh -c groupadd -g ${gid} ${group}     && useradd -d "${JENKINS_AGENT_HOME}" -u "${uid}" -g "${gid}" -m -s /bin/bash "${user}"' returned a non-zero code: 1
[root@openpower8 ubuntu-s390x]# 
```

**Review by Codysseus — APPROVED** [review]
Tested on s390x-docker2

**Pull request merged**